### ISSUE TITLE
Image re-import sometimes doesn't work and keeps old textures.

### ISSUE THREAD

**Issue by avencherus** [godotengine/godot#22747]
**Windows 10 64-bit Alpha 3.1 8068d02**

When replacing PNG files, sometimes Godot now doesn't update and reimport the new image.  Even when forcing it, it still has some display issues that aren't corrected, and **the old texture may still appear even when restarting the editor**?

Hard to say exactly how to reproduce this, but it happened at least twice when:

1. I created a new scene, dragged a PNG from the file manager onto the canvas as a Sprite.
2. Replaced the source image in the folder from Windows Explorer

Below I had originally dragged a PNG of an X11 icon, then replace it with a Win10 icon.

## Overwrote the image in Windows Explorer
(Changed the os.png from the X11 one to a Win10 version.  No change on the canvas or in the file manager.)
![godot_master_2018-10-05_05-58-03](https://user-images.githubusercontent.com/13004169/46514047-4255bc80-c864-11e8-8eb3-4ef69bbdfb78.png)

## Manually re-import sort of works, but is cut off.
(At one point it was saying it was 54x60 or something, the canvas texture shows that at least.)

![godot_master_2018-10-05_05-59-17](https://user-images.githubusercontent.com/13004169/46514099-7f21b380-c864-11e8-8e60-f488bf581495.png)

## Still shows the old icon in the file manager.
![godot_master_2018-10-05_05-59-38](https://user-images.githubusercontent.com/13004169/46514201-3ae2e300-c865-11e8-8a3b-2b23b14d3401.png)

## Overwrote the Win icon back to the X11 and restarted the editor.
(The textures for the windows icon were overwritten with the X11 one, but they still remain in the scenes?)

![godot_master_2018-10-05_06-08-09](https://user-images.githubusercontent.com/13004169/46514236-77aeda00-c865-11e8-9412-69e8d46346f1.png)

**Comment by akien-mga** [comment]
That's a feature obviously, you're not allowed to replace Linux by Windows.

**Comment by akien-mga** [comment]
Could be a regression from a recent warning fix in `core/image.cpp` or `drivers/png/image_loader_png.cpp`:
- f48ee838e
- dec20a987 (especially this one fixed a mipmaps "bug", but maybe this bug was expected behaviour)
- 95131e6f

**Comment by avencherus** [comment]
> That's a feature obviously, you're not allowed to replace Linux by Windows.

It did warn me after all!  XD

![chrome_2018-10-05_12-02-23](https://user-images.githubusercontent.com/13004169/46526307-92e60d80-c896-11e8-963c-7c96fed6d846.png)

**Comment by avencherus** [comment]
@akien-mga Thanks for sharing the commits.  I'll see if reverting them and building in the morning will keep me working from the head.

I at least tried out reverting dec20a9

It improves, but has some issue still.  The reimport seems responsive again, but the texture sizes are wrong on the sprite, and will refresh when you drag it.

## Replacing blue texture with larger red texture.
![ezgif com-gif-maker](https://user-images.githubusercontent.com/13004169/46560486-a1671000-c8fc-11e8-8428-a2076c5fd393.gif)

## Replacing red texture back to the smaller blue one.
![ezgif com-video-to-gif 1](https://user-images.githubusercontent.com/13004169/46560497-aaf07800-c8fc-11e8-871b-f73f4b5e80fc.gif)


## File Drag & Drop Issues

I also tried dragging and dropping the PNG from Windows explorer to the file manager panel in the editor, and got this error.  (Not experiencing this in my previous build.)

![explorer_2018-10-06_00-01-58](https://user-images.githubusercontent.com/13004169/46560539-d2474500-c8fc-11e8-9c63-9f73889d34de.png)

**Comment by avencherus** [comment]
Not much luck reverting the others.  Ran into a merge conflict, and boy this is an area I know nothing about.  X)

We're doing a lot of animation stuff this week and lots of texture importing to be done, so I'll be hanging back in 874e3b4 for the time being.

**Comment by akien-mga** [comment]
I confirm that dec20a98 caused the reimport regression, I'll dig further to see what part of the `core/image.cpp` changes are to blame, and fix that.

I can also confirm that there is still an issue with updating the size, but I don't know yet what caused that.

**Comment by akien-mga** [comment]
> I can also confirm that there is still an issue with updating the size, but I don't know yet what caused that.

@avencherus I still get this issue in 874e3b4, so it doesn't seem to be a recent regression.

**Comment by avencherus** [comment]
@akien-mga Oh you're right, the texture resize issue is present there as well.

I noticed the drag and drop problem is something else entirely.  I just wasn't getting it before until I came back to do some retests.  Seems like it has something to do with where the file is dropped.

I will go explore some older versions and see what I see, and try to figure out how to precisely reproduce the drag and drop issue.

Seems like this is maybe three different issues, I just happened to incur them all at the same time.
